### PR TITLE
[feat] `adjust_sigmoid` and `adjust_log` initial implementation

### LINKED ISSUE #1619: Add `adjust_sigmoid` to kornia.enhance
We would like to have a consistent functionality against scitkit-image `adjust_sigmoid`

See: https://scikit-image.org/docs/stable/api/skimage.exposure.html#adjust-sigmoid

## Motivation

Useful for image manipulation and  to compose complex augmentations 

______________________________________________________________________

#### Consider also to contribute to Kornia universe projects :)

<sub>

- [**Tutorials**](https://github.com/kornia/tutorials): our repository containing the tutorials.

</sub>

### PR DESCRIPTION
**Pull request by dobosevych**
#### Changes

Fixes #1619 #1618 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by dobosevych** [comment]
I would add #1618 in the same PR and tests

**Comment by dobosevych** [comment]
Fixes https://github.com/kornia/kornia/issues/1618

File: kornia/enhance/adjust.py [hunk 1]
```diff
@@ -529,6 +529,76 @@ def adjust_brightness_accumulative(image: Tensor, factor: Union[float, Tensor],
...
+
+    Example:
+        >>> x = torch.ones(1, 1, 2, 2)
+        >>> adjust_sigmoid(x, gain=0)
+        tensor([[[[0.5000, 0.5000],
+                  [0.5000, 0.5000]]]])
+    """
+    KORNIA_CHECK_IS_TENSOR(image, "Expected shape (*, H, W)")
+
+    if inv:
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:558]: I think it would make sense to have `adjust_sigmoid` and `adjust_sigmoid_inverse`. I'm not a fan of free functions that by a flag behaves totally different.

File: kornia/enhance/adjust.py [hunk 2]
```diff
@@ -529,6 +529,76 @@ def adjust_brightness_accumulative(image: Tensor, factor: Union[float, Tensor],
...
+        img_adjust = (1 - 1 / (1 + torch.exp(gain * (cutoff - image))))
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:559]:
```suggestion
        img_adjust = (1 - 1 / (1 + (gain * (cutoff - image)).exp()))
```
more pythinic toward supporting other frameworks

File: test/enhance/test_adjust.py [hunk 3]
```diff
@@ -637,6 +637,71 @@ def test_gradcheck_accumulative(self, device, dtype):
...
+                 [[0.07585, 0.07585], [0.07585, 0.07585]]],
+                [[[0.00669, 0.00669], [0.00669, 0.00669]], [[0.1192, 0.1192], [0.1192, 0.1192]],
+                 [[0.7310, 0.7310], [0.7310, 0.7310]]],
+            ],
+            device=device,
+            dtype=dtype,
+        )  # 2x3x2x2
+
+        f = kornia.enhance.AdjustSigmoid()
+        assert_close(f(data), expected)
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:669]: can you add gradcheck, and/or onxx-jit tests ?

**Review by edgarriba — APPROVED** [review]
can you also add the functions to the docs? Please, @shijianjian @ducha-aiki review

File: test/enhance/test_adjust.py (hunk 3 above)
Comment (dobosevych) [reply test/enhance/test_adjust.py:670]: Added, can you check? I suppose that all such tests should be marked with specific marks.

File: kornia/enhance/adjust.py (hunk 2 above)
Comment (dobosevych) [reply kornia/enhance/adjust.py]: Made more pythinic

File: kornia/enhance/adjust.py (hunk 1 above)
Comment (dobosevych) [reply kornia/enhance/adjust.py:558]: Not sure about that, as we would have four functions instead of 2. I understand that you are pointing to the Clean Code suggestion, but personally, I for experiments prefer flag. I can modify that if you think that it would be better.

File: test/enhance/test_adjust.py (hunk 3 above)
Comment (edgarriba) [reply test/enhance/test_adjust.py:670]: marks are not really required

File: kornia/enhance/adjust.py (hunk 1 above)
Comment (edgarriba) [reply kornia/enhance/adjust.py:558]: i'd like to have all this set of low level operators with very specific naming of what they do and possibly with proper references. We had similar discussion during the contrast/brightness design specially because there's a trend of just copying behaviors from other libraries that don't follow the theoretical definition. In that case we decided doing something similar to here with [`adjust_contrast_with_mean_subtraction`](https://github.com/kornia/kornia/blob/master/kornia/enhance/adjust.py#L357) /cc @shijianjian @ducha-aiki
